src/usyms.cpp

```diff
@@ -47,8 +87,71 @@ void Usyms::cache_bcc(const std::string &elf_file)
       pid_sym_[pid] = bcc_symcache_new(pid, &get_symbol_opts());
 }
 
+#ifdef HAVE_BLAZESYM
+struct blaze_symbolizer *Usyms::create_symbolizer() const
+{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+  blaze_symbolizer_opts opts = {
+    .type_size = sizeof(opts),
+    .demangle = config_.get(ConfigKeyBool::cpp_demangle),
+  };
+#pragma GCC diagnostic pop
+
+  return blaze_symbolizer_new_opts(&opts);
+}
+
+void Usyms::cache_blazesym(const std::string &elf_file)
+{
+  auto cache_type = config_.get(ConfigKeyUserSymbolCacheType::default_);
+  if (cache_type == UserSymbolCacheType::none)
+    return;
+
+  if (symbolizer_ == nullptr) {
+    symbolizer_ = create_symbolizer();
+    if (symbolizer_ == nullptr)
+      return;
+  }
+
+  // preload symbol table for executable to make it available even if the
+  // binary is not present at symbol resolution time
```

Comment:
"binary is not present" meaning binary is deleted or process is not running?

Reply:
It's just a copy of the same blurb from the bcc path -- I anticipated its upcoming removal and figured we'd loose it otherwise. Anyway, it refers to the binary no longer being available on the file system, i.e., having been deleted.